synapse/handlers/presence.py

```diff
@@ -525,41 +520,38 @@ def update_external_syncs_row(
                 process_presence.discard(user_id)
 
             if updates:
-                yield self._update_states(updates)
+                await self._update_states(updates)
 
             self.external_process_last_updated_ms[process_id] = self.clock.time_msec()
 
-    @defer.inlineCallbacks
-    def update_external_syncs_clear(self, process_id):
+    async def update_external_syncs_clear(self, process_id):
```

Comment:
looks like ReplicationStreamer calls this without an `await` or a `run_as_background_process`.

Reply:
Good spot! I wonder if we can make the tests detect unawaited coroutines somehow or something (ala https://bugs.python.org/issue30491)